core/src/test/scala/integration/kafka/api/PlaintextEndToEndAuthorizationTest.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.api
+
+import kafka.api.{EndToEndAuthorizationTest, ZkSasl}
+import org.apache.kafka.common.config.internals.BrokerSecurityConfigs
+import org.apache.kafka.common.network.ListenerName
+import org.apache.kafka.common.protocol.SecurityProtocol
+import org.apache.kafka.common.security.auth.{AuthenticationContext, KafkaPrincipal, KafkaPrincipalBuilder}
+import org.junit.Before
+
+// This test case uses a separate listener for client and inter-broker communication, from
+// which we derive corresponding principals
+object PlaintextEndToEndAuthorizationTest {
+  class TestClientPrincipalBuilder extends KafkaPrincipalBuilder {
+    override def build(context: AuthenticationContext): KafkaPrincipal = {
```

Comment:
May be worth checking that this is a PlainAuthenticationContext with a loopback address as the client address?

Reply:
I think it's worth checking the authentication context type, but I'm not sure how valuable it is to verify the client address since it will be the same for both listeners. Perhaps there is a unit test we are missing where this would make more sense?

Reply:
I couldn't find a test which checked that the client address was populated at all. Agree that this is not ideal since it will be the same for both listeners. But checking here is still useful since it would verify that
1. client address is populated in `AuthenticationContext`
2. address was obtained after the socket was in connected state (it would be null otherwise)

Reply:
Maybe a null check would be sufficient? I was hesitant about the loopback check since I'm not sure if that would be reliable across different environments.